### PR TITLE
Feat/facilitie component

### PR DESCRIPTION
**Pull request by santosluizfelipe**
Added  the double tap and long press functionality, changed the background color, added the  clear all and next buttons and changed the grid from 3 to 2 icons per row

### CODE REVIEW

**Comment by siqbal181** [comment]
@santosluizfelipe some merge conflicts to resolve :)

**Comment by pablisch** [comment]
@santosluizfelipe can we look at this together tomorrow (Tuesday)? There are some names I'd like to work through with you. I completely understand how 'updatedDoubleFacilityList' came about (mostly because I am part of the problem) and what it means, but I think we can make names clearer for anyone looking at this with fresh eyes.

**Comment by pablisch** [comment]
Hi @santosluizfelipe, I have not fully reviewed this as changes needed may well have a knock-on effect but there are a few manual testing things to sort. Some needed and some useful.

- [ ] Reset should clear all selections (currently only clears wanted facilities)
- [ ] Continue/Next/Submit should submit when there are only required facilities (currently needs there to be at least one wanted facility)
- [ ] Submit should log wanted and required facilities (currently only logs wanted facilities as 'facilities')
- [ ] All postman tests fail. These all pass when I run them and they seem to pass in main. Can't see why your branch should make any different to Postman API tests. Just wondered if you had any idea as this is a mystery to me.

**Comment by pablisch** [comment]
This is also a good time to remove react allergens if we are never using them as they will be there forever otherwise.

**Comment by santosluizfelipe** [comment]
> Hi @santosluizfelipe, I have not fully reviewed this as changes needed may well have a knock-on effect but there are a few manual testing things to sort. Some needed and some useful.
> 
> * [ ]  Reset should clear all selections (currently only clears wanted facilities)
> * [ ]  Continue/Next/Submit should submit when there are only required facilities (currently needs there to be at least one wanted facility)
> * [ ]  Submit should log wanted and required facilities (currently only logs wanted facilities as 'facilities')
> * [ ]  All postman tests fail. These all pass when I run them and they seem to pass in main. Can't see why your branch should make any different to Postman API tests. Just wondered if you had any idea as this is a mystery to me.

the postman tests might be because i did not send any data to the backend it is only frontend changes

**Comment by pablisch** [comment]
> the postman tests might be because i did not send any data to the backend it is only frontend changes

They should be totally independent. There was never any requests to the backend. It might be a CI issue but weirdly has so far only happened on this branch so I thought I would ask. We might just have to disable that part of CI until we understand why this is happening.

**Comment by pablisch** [comment]
I made a branch from your branch, made a PR for it and now the tests pass. NO clue why but that is one step closer.

**Comment by pablisch** [comment]
Sorry Luiz, I didn't have a chance to look properly but as far as I got, I could see that facilities were able to be in both facility arrays at the same time when each should only appear in one.
And really small thing, but could you change the facility log to be wanted facilities to align with the new names.

File: frontend/src/components/FacilityForm/FacilityForm.js [hunk 1]
```diff
@@ -61,16 +116,18 @@ const FacilityForm = () => {
...
+                  isDoubleSelected={requiredFacilities.includes(facility.name)}
                 />
               </View>
             ))}
           </View>
         </View>
         <View>
           <Pressable>
-            <Button onPress={() => handleSubmit()} title={"The old button!"} />
+            <Button onPress={() => handleSubmit()} title={"Continue"} />
```
Comment (pablisch) [inline frontend/src/components/FacilityForm/FacilityForm.js:130]: I deliberately changed this to make it clear it was no longer 'the' button. Please undo this change or change to anything that indicates it is the old button that makes sense to you. There is a ticket for moving this functionality to the new buttons in the menu bar.

File: frontend/src/components/Button/Button.js [hunk 2]
```diff
@@ -2,14 +2,15 @@ import React from "react";
 import { TouchableOpacity, Text } from "react-native";
 import { styles } from "./ButtonStyles";
 
-const Button = ({ onPress, title }) => {
+const Button = ({ onPress, title, isPrimary }) => {
```
[the remark this answers is not in the document]
Comment (pablisch) [reply frontend/src/components/Button/Button.js]: @santosluizfelipe at first glance it looks like the Button issue remains so I have not reviewed the rest of the PR yet. Happy to discuss the issue and why it is an issue on a call but everything in the comment above about Buttons remains. Can I suggest that if everything else is good, you make a branch of this branch to retain the code related to the Button for possible future use in #82 and then remove all Button changes from this branch. I would really like to get this ticket into main.

If I am missing something here, please let me know.

[the remark this answers is not in the document]
Comment (santosluizfelipe) [reply frontend/src/components/Button/Button.js]: @pablisch  I removed the isPrimary from the button, not sure why is showing up here still, could you pull the code and double check because in my local code is sorted.

[the remark this answers is not in the document]
Comment (pablisch) [reply frontend/src/components/Button/Button.js]: Ah, I see. I was talking about **all** the Button changes but the only reversion I can see is removing the isPrimary prop which actually causes a problem as it is elsewhere and now undefined. I think the work you did may be a good start to the re-usability issue but leaving it as code that is not used anywhere may actually create more work for anyone taking #82 which is what I am trying to avoid in addition to a principle of only passing in functional code into main. 

I have not run the branch yet and need to be elsewhere for a couple of hours. Honestly, I am not concerned what the button looks like at the moment as that will be dealt with when the Button is looked at properly.

[the remark this answers is not in the document]
Comment (pablisch) [reply frontend/src/components/Button/Button.js]: Hi @santosluizfelipe, I had a chance to run this branch now and I get an error due to the undefined isPrimary mentioned above. Not sure why you do not get the same?

I reverted to the previous Button code and with that error resolved, I'm afraid to say there is still the issue of pressing a required facility does not deselect it but makes it a wanted facility.

**Review by pablisch — APPROVED** [review]
🎉
